llvm/lib/Target/RISCV/RISCVInstrInfoV.td

```diff
@@ -1677,8 +1677,9 @@ let Predicates = [HasVInstructions] in {
 let Constraints = "@earlyclobber $vd", RVVConstraint = Vrgather in {
 defm VRGATHER_V : VGTR_IV_V_X_I<"vrgather", 0b001100>;
 def VRGATHEREI16_VV : VALUVV<0b001110, OPIVV, "vrgatherei16.vv">,
-                      SchedBinaryMC<"WriteVRGatherVV", "ReadVRGatherVV_data",
-                                    "ReadVRGatherVV_index">;
+                      SchedBinaryMC<"WriteVRGatherVVEEW",
```

Comment:
I'm unsure what EEW is meant to convey here. Every operand of every vector instruction has an EEW. For vrgather.vv, for all operands EEW=SEW. For vrgatherei16.vv, the index operand has EEW=16 and the data and index operands have EEW=SEW.

Reply:
It is meant to convey that it is used for `vrgatherei16.vv` and not `vrgather.vv`. Would it be better if we did `WriteVRGatherEI16VV`?

Reply:
I put EEW at the end because of `VPseudoVGTR_VV_EEW` naming

Reply:
> It is meant to convey that it is used for vrgatherei16.vv and not vrgather.vv. Would it be better if we did WriteVRGatherEI16VV

I think so

> I put EEW at the end because of VPseudoVGTR_VV_EEW naming

Looks like that takes `eew` as a template parameter. So we should use that parameter to form the scheduler class name. Or maybe we should change the name and drop that parameter?

Reply:
What do you think about `VPseudoVGTR_VV_EEW<int eew, ...>` -> `VPseudoVGTREI16_VV<int eew, ...>`

Reply:
I would remove `int eew` as a parameter if you put EI16 in the name.

Reply:
998fc2c2a6a3bc0758c7964a988fa0dea5a4c620